### ISSUE TITLE
Update product screens to reference PB

### ISSUE THREAD

**Issue by Abi-Nada** [cds-snc/platform-forms-client#2371]
### User Story:

As a user, I want to be able to set "Protected B" as a data classification within the product, so that it correctly reflects the sensitivity of the information being gathered, and form responses are delivered to the right place.

### Acceptance Criteria:
- [x] The Settings menu includes an option to select the data classification level, and "Protected B" is available as a selectable option.
- [x] The response delivery method is adjusted based on the selected data classification level, adhering to the necessary requirements (Protected B means you can't receive responses through emails).
- [ ] Functionality is implemented as per design wireframes

**Comment by timarney** [comment]
PR branch
https://github.com/cds-snc/platform-forms-client/pull/2374


Preview URL
https://yhahfutujmygvpjkrfyfwkjfya0ohoyt.lambda-url.ca-central-1.on.aws/form-builder/settings/cljsp5e69000aye66gp1fx67l

<img width="840" alt="Screenshot 2023-07-10 at 2 28 46 PM" src="https://github.com/cds-snc/platform-forms-client/assets/62242/0d08817c-ffdc-433c-b568-955ec4b26811">

**Comment by timarney** [comment]
Preview URL 

https://yhahfutujmygvpjkrfyfwkjfya0ohoyt.lambda-url.ca-central-1.on.aws/

**Comment by timarney** [comment]
Folks can review the Preview Url --- this work is on hold as we can't launch the content / merge until we have Protected B status.

**Comment by anikbrazeau** [comment]
Done! ✅ 🚀
